cfgov/unprocessed/apps/owning-a-home/js/explore-rates/tab.js

```diff
@@ -1,19 +1,53 @@
-'use strict';
+let tabGroups;
+let tabLinks;
+let tabContents;
 
-var $ = require( 'jquery' );
+/**
+ * Initialize the functionality of a group of tabs with content.
+ */
+function init() {
+  tabGroups = document.querySelectorAll( '.tabs-layout' );
 
-$( '.tab-link' ).click( function( evt ) {
-  var $tabs = $( '.tab-list' ),
-      $tabLi = $( this ).parent( '.tab-list' ),
-      $tabContent = $( '.tab-content' ),
-      current = $( this ).attr( 'href' );
+  tabGroups.forEach( tabGroup => {
+    tabLinks = tabGroup.querySelectorAll( '.tab-link' );
+    tabContents = tabGroup.querySelectorAll( '.tab-content' );
 
-  $tabs.removeClass( 'active-tab' );
-  $tabLi.addClass( 'active-tab' );
+    tabLinks.forEach( tabLink => {
+      _bindTabLink( tabGroup, tabLink, tabContents );
+    } );
 
-  $tabContent.hide();
-  $( current ).show();
+    // Hide all but the first tab.
+    for ( let i = 1; i < tabContents.length; i++ ) {
+      tabContents[i].classList.add( 'u-hidden' );
+    }
+  } );
+}
 
-  evt.preventDefault();
+/**
+ *
+ * @param {HTMLNode} tabGroup - HTML element parent of tabs and tabs-content.
+ * @param {HTMLNode} tabLink - HTML element for links in tabs.
+ */
+function _bindTabLink( tabGroup, tabLink, tabContents ) {
+  tabLink.addEventListener( 'click', _tabLinkClicked );
+  const tabs = tabGroup.querySelectorAll( '.tab-list' );
 
-} );
+  /**
+   * Handle a click of a tab.
+   * @param {MouseEvent} evt - Event object for a click on a tab.
+   */
+  function _tabLinkClicked( evt ) {
+    evt.preventDefault();
+
+    const target = evt.target;
+    const tabLi = target.parentNode;
+    const current = tabGroup.querySelector( target.getAttribute( 'href' ) );
+
+    tabs.forEach( tab => tab.classList.remove( 'active-tab' ) );
```

Comment:
Use the dom-tools library to ensure we don't thrash.

Reply:
Are we sure this is needed here since it's only two tabs? dom-tools requires a second dom lookup because the method signature for `removeClass` is a selector not a HTMLNode.

Reply:
So you can pass in a Node or selector like: `DT.removeClass( tab, 'active-tab' ).

Reply:
Oh I see, it passes it to `isEl`. Would we want this for all CSS class manipulation, such as `tabLi.classList.add( 'active-tab' )` on the next line? Any way to test whether it's needed? I'll merge this for now to get the tests in and we can adjust.

Reply:
Perhaps we should jsdoc the `selector` parameter's type. That's where it's useful 😁